src/main/kotlin/ca/kittle/util/StringUtils.kt

```diff
@@ -1,10 +1,43 @@
 package ca.kittle.util
 
+import kotlinx.serialization.ExperimentalSerializationApi
 import kotlinx.serialization.json.Json
+import java.security.SecureRandom
+import java.time.Instant
+import java.util.*
 
+@OptIn(ExperimentalSerializationApi::class)
 val json =
     Json {
         isLenient = true
         ignoreUnknownKeys = true
         encodeDefaults = true
+        explicitNulls = false
     }
+
+private val random = SecureRandom()
+
+fun randomUUIDv7(): UUID {
```

Comment:
**suggestion:** Consider using an existing library for UUID generation

While the custom UUIDv7 implementation is interesting, it might be more robust and maintainable to use an existing, well-tested library for UUID generation. If there's a specific reason for this custom implementation, please document it.